probe-rs/src/architecture/riscv/dtm/dtm_access.rs

```diff
@@ -0,0 +1,102 @@
+use crate::architecture::riscv::communication_interface::RiscvError;
+use crate::probe::{CommandResult, DeferredResultIndex};
+use crate::{DebugProbe, DebugProbeError, Probe};
+use bitfield::bitfield;
+use std::fmt;
+use std::time::Duration;
+
+pub trait DtmAccess: Send + fmt::Debug {
```

Comment:
This still feels very JTAG specific to me. E.g. the `DmiOperation` seems to be defined just in the context of the `JTAG Debug Transport Module` in the debug spec.

It would be great if an abstraction on the level of just register reads and writes was possible, so that I can just use something `interface.read(0x10)` to read the `dmcontrol` register.

Reply:
I agree, this should be abstracted away. I will look into it next week.

Reply:
DmiOperation is now only in jtag_dtm.rs

I expect it will be moved later into some shared module as WCH RVSWD uses it as well, in the exact same way.